lib/utils.js

```diff
@@ -131,6 +131,44 @@ async function updateFromOrigin(mainUrl, repo, branch) {
   }
 }
 
+async function connectToRemote(repo, remote = 'origin') {
+  const repoOrigin = await repo.getRemote(remote)
+
+  await repoOrigin.connect(Git.Enums.DIRECTION.FETCH, {
+    credentials: promptForCredentials,
+    certificateCheck() {
+      return 1
+    }
+  })
+
+  return repoOrigin
+}
+
+async function headExistInRemote(repo, remote, head) {
+  const origin = await connectToRemote(repo, remote)
+  const references = await origin.referenceList()
+
+  return references.filter(reference => reference.name() === head)[0]
```

Comment:
*.find* will terminate once it finds a match. Does that help in this context?
```suggestion
  return references.find(reference => reference.name() === head)
```